src/sql/run.py

```diff
@@ -106,13 +106,18 @@ def __init__(self, sqlaproxy, config):
         self.keys = {}
         if sqlaproxy.returns_rows:
             self.keys = sqlaproxy.keys()
-            if config.autolimit:
+            if isinstance(config.autolimit, int) and config.autolimit > 0:
```

Comment:
why do we need this? this would break existing compatibility (setting None and setting autolimit to 0 should display all values) - it's a bit counterintuitive but we inherited this behavior from ipython-sql

Reply:
Since I removed the hardcoded configuration from `SqlCmdMagic` there are some missing default configurations (`autolimit` and `style`). In this case, `config.autolimit` and `config.style` return `<LazyConfigValue>`.

I added the `config.autolimit > 0` since according to one of the tests if autolimit is 0 we should return everything.

Reply:
ah, I see what you're saying! we have a design problem here.

the configuration is attached to the `%sql` magic, but ideally we want the config to be accessible to all magics. I'm unsure if this is possible so I think for now the best thing to do is to create another version of `run` that doesn't take the config argument (we won't be able to provide the autolimit feature when running the profiling but that's fine)

I remember suggesting creating a `raw_run` function already but I can't remember if it was in the ggplot PR or in a different one that @tonykploomber is working on.